### DISCUSSION TITLE
Error with getReference and Attribute in Entity in Symfony 7 and Doctrine 3

### DISCUSSION THREAD

**Discussion by ebootingdev** [symfony/symfony#59215] [Q&A]
Hi everyone,

I have this case, which works fine:

<code>
    //Entity
    /**
     * @var MainState
     */
    #[ORM\JoinColumn(name: 'id_state', referencedColumnName: 'id')]
    #[ORM\ManyToOne(targetEntity: MainState::class)]
    private $state;

    //Controller
    $entity->setState($this->entityManager->getReference(MainState::class, 1));
</code>

But when I do it this way:

<code>
   //Entity
   #[ORM\JoinColumn(name: 'id_state', referencedColumnName: 'id')]  
   #[ORM\ManyToOne(targetEntity: MainState::class)]
    private ?MainState $state = null;

   //Controller
    This line doesn't work and throws an error:
    $entity->setState($this->entityManager->getReference(MainState::class, 1));
    PHP Fatal error:
    Uncaught TypeError: Cannot assign __PHP_Incomplete_Class to property App\\Entity\\Entity::$state of type ?App\\Entity\\MainState
  
    I have to replace:
    $entity->setState($this->entityManager->getReference(MainState::class, 1));
    with:
    $entity->setState($this->repositoryMainState()->find(1));

    Why doesn't getReference work?
    [composer.json](https://github.com/user-attachments/files/18143710/composer.json)
  

</code>

**Comment by MatTheCat** [comment]
Hello,

AFAIU this has nothing to do with Doctrine or attributes: the working code does not type `$state` so you can set it to an instance of `__PHP_Incomplete_Class` without PHP complaining, even if this is certainly wrong. Your issue stems from a failed unserialization so you should check that way.

**Reply by ebootingdev** [reply]
But if you notice, the error occurs when I change this line:
private $state;
to
private ?MainState $state = null; (specifying that the variable is of type ?MainState).

And in the controller, keeping this line:
$entity->setState($this->entityManager->getReference(MainState::class, 1));
with the getReference() method.

For it to work correctly, I have to change this line:
$entity->setState($this->entityManager->getReference(MainState::class, 1));
to
make select to repository before
$entity->setState($this->repositoryMainState()->find(1));.

In other words, passing the variable $state as ?MainState causes getReference not to work.
It’s worth mentioning that I’m using PHP 8.3.14.

**Comment by ebootingdev** [comment]
"I found the error:
https://stackoverflow.com/questions/965611/forcing-access-to-php-incomplete-class-object-properties
I was calling a session_start() beforehand, and that was causing the error."
